ext/src/ruby_api/engine.rs

```diff
@@ -168,8 +171,9 @@ impl Engine {
     /// @return [String] Binary String of the compiled module.
     /// @see Module.deserialize
     pub fn precompile_module(&self, wat_or_wasm: RString) -> Result<RString, Error> {
-        self.inner
-            .precompile_module(unsafe { wat_or_wasm.as_slice() })
+        let (wat_or_wasm, _guard) = unsafe { wat_or_wasm.as_locked_slice() }?;
```

Comment:
First time I come across [`rb_str_locktmp`](https://github.com/ruby/ruby/blob/31371b2e24b03ccb0a03b622faf8c65e6cf6a31a/include/ruby/internal/intern/string.h#L615) (thanks!).

IIUC:
- the lock ensures the string won't be mutated by another Ruby thread
- the RString's value is on the stack so it won't get GC'd nor moved/compacted

Do I get this right?

Reply:
Yes and won't be "tampered with" (according to docs)

>  Obtains a "temporary  lock" of the string.  This  advisory locking mechanism
> prevents other  cooperating threads from  tampering the receiver.   The same
> thing could be done via freeze mechanism,  but this one can also be unlocked
> using rb_str_unlocktmp().